minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterControl.java

```diff
@@ -111,6 +127,53 @@ public void adminStopAll() throws IOException {
     }
   }
 
+  private static TExternalCompactionList getRunningCompactions(ClientContext context)
+      throws TException {
+    Optional<HostAndPort> coordinatorHost =
+        ExternalCompactionUtil.findCompactionCoordinator(context);
+    if (coordinatorHost.isEmpty()) {
+      throw new TTransportException("Unable to get CompactionCoordinator address from ZooKeeper");
+    }
+    CompactionCoordinatorService.Client client = ThriftUtil.getClient(
+        new CompactionCoordinatorService.Client.Factory(), coordinatorHost.get(), context);
+    try {
+      TExternalCompactionList running =
+          client.getRunningCompactions(TraceUtil.traceInfo(), context.rpcCreds());
+      return running;
+    } finally {
+      ThriftUtil.returnClient(client, context);
+    }
+  }
+
+  public synchronized void startCoordinator(Class<? extends CompactionCoordinator> coordinator)
+      throws IOException {
+    if (coordinatorProcess == null) {
+      coordinatorProcess = cluster
+          ._exec(coordinator, ServerType.COMPACTION_COORDINATOR, new HashMap<>()).getProcess();
+      UtilWaitThread.sleep(1000);
+      // Wait for coordinator to start
+      TExternalCompactionList metrics = null;
+      while (null == metrics) {
+        try {
+          metrics = getRunningCompactions(cluster.getServerContext());
+        } catch (Exception e) {
+          UtilWaitThread.sleep(250);
+        }
+      }
+    }
+  }
+
+  public synchronized void startCompactors(Class<? extends Compactor> compactor, int limit,
+      String queueName) throws IOException {
+    synchronized (compactorProcesses) {
+      int count = 0;
+      for (int i = compactorProcesses.size();
+          count < limit && i < cluster.getConfig().getNumCompactors(); i++, ++count) {
+        compactorProcesses.add(cluster.exec(compactor, "-q", queueName).getProcess());
+      }
```

Comment:
This isn't shorter, but I have found the IntStream semantics to be more readable in some cases. The following might be a suitable replacement, but it is a bit longer. It'd be shorter if there weren't any checked exceptions thrown.
```suggestion
      IntStream.range(compactorProcesses.size(), cluster.getConfig().getNumCompactors()).limit(limit).forEach(i -> {
        try {
          compactorProcesses.add(cluster.exec(compactor, "-q", queueName).getProcess());
        } catch (IOException e) {
          throw new UncheckedIOException(e);
        }
      });
```

Reply:
I left this as-is in [851c2c3]. The old style for loop is well-known. The new stream-style is only more readable if you are familiar with it.

Reply:
That's fine. I think the main issue is comprehensibility around the juggling of multiple loop variables. Here's another way this could be made more intuitive:

```suggestion
      int count = Math.min(limit, cluster.getConfig().getNumCompactors() - compactorProcesses.size());
      for (int i = 0; i < count; i++) {
        compactorProcesses.add(cluster.exec(compactor, "-q", queueName).getProcess());
      }
```

Reply:
Applied change in 47981eb